File: LanguAI.Frontend/src/api/models/save-interaction-request-view-model.ts
/* tslint:disable */
/* eslint-disable */
import { InteractionEnum } from '../models/interaction-enum';
export interface SaveInteractionRequestViewModel {
  content?: string | null;
  id?: number | null;
  interactionType?: InteractionEnum;
  isDeleted?: boolean;
  parentInteractionId?: number | null;
  postId?: number | null;
  userId?: number;
}

File: LanguAI.Backend/LanguAI.Backend/ViewModels/Interaction/SaveInteractionRequestViewModel.cs
using LanguAI.Backend.Core.Enums;

namespace LanguAI.Backend.ViewModels.Interaction;

public class SaveInteractionRequestViewModel
{
    public int? Id { get; set; }

    public InteractionEnum InteractionType { get; set; } = InteractionEnum.Like;

    public int UserId { get; set; }

    public int? PostId { get; set; }

    public int? ParentInteractionId { get; set; }

    public string Content { get; set; }

    public bool IsDeleted { get; set; } = false;
}
